ruma-api-macros/src/api.rs

```diff
@@ -428,18 +428,25 @@ impl Parse for RawMetadata {
 }
 
 pub struct RawRequest {
+    pub attributes: Vec<Attribute>,
     pub request_kw: kw::request,
     pub fields: Vec<Field>,
 }
 
 impl Parse for RawRequest {
     fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
+        // If there are attributes to parse otherwise instead of failing
+        // emit empty vec.
+        let attributes =
+            if let Ok(attrs) = input.call(Attribute::parse_outer) { attrs } else { vec![] };
```

Comment:
I think this kind of thing is usually done using `peek` rather than taking an alternative branch if parsing fails. I'm not sure what this code would do on unexpected input (i.e. tokens that are none of [attribute, request_ke]).

Reply:
This works `let attributes = input.call(Attribute::parse_outer)?;`

I was wondering about not advancing the "cursor" and stuff I originally did have `peek`, but the syn docs say to use `.call` for attributes but now it will fail when it's supposed to I think.